airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/xmin/XminCtidUtils.java

```diff
@@ -60,6 +70,12 @@ public static StreamsCategorised categoriseStreams(final StateManager stateManag
     return new StreamsCategorised(new CtidStreams(streamsForCtidSync, statesFromCtidSync), new XminStreams(streamsForXminSync, statesFromXminSync));
   }
 
+  @VisibleForTesting
+  static boolean shouldPerformFullSync(final XminStatus currentXminStatus, final JsonNode streamState) {
```

Comment:
I think it may be better to have all wraparound calculations in a single class

Reply:
+1 we can just move to `PostgresXminHandler`